CMakeLists.txt

```
@@ -3,7 +3,7 @@
 
 cmake_minimum_required(VERSION 3.16)
 
-project(iDynTree VERSION 10.3.0
+project(iDynTree VERSION 10.4.0
```

```suggestion
project(iDynTree VERSION 11.0.0
```

Let's bump major version due to the ABI change.